pkg/controller/nodemaintenance/nodemaintenance_controller.go

```diff
@@ -254,6 +256,37 @@ func (r *ReconcileNodeMaintenance) Reconcile(request reconcile.Request) (reconci
 	return reconcile.Result{}, nil
 }
 
+
+func setOwnerRefToNode(instance *nodemaintenanceapi.NodeMaintenance, node *corev1.Node) (bool) {
+
+	objectRefIsSet := false
+	for _, ref := range instance.ObjectMeta.GetOwnerReferences() {
+		if ref.APIVersion == node.TypeMeta.APIVersion && ref.Kind == node.TypeMeta.Kind && ref.Name == node.ObjectMeta.GetName() && ref.UID == node.ObjectMeta.GetUID() {
+			objectRefIsSet = true
```

Comment:
You can just `return false` here, and remove the `objectRefIsSet` var and the following `if` condition

Reply:
done